pkgs/npm/src/cli.ts

```diff
@@ -24,11 +24,15 @@ function getPathToExecutable(): string {
 // in the user PATH to the cli.js to execute. The symlink will name the same as
 // the package name i.e. defang.
 function run(): void {
-  const args = process.argv.slice(2);
-  const processResult = child_process.spawnSync(getPathToExecutable(), args, {
-    stdio: "inherit",
-  });
-  process.exit(processResult.status ?? 0);
+  try {
+    const args = process.argv.slice(2);
+    const processResult = child_process.spawnSync(getPathToExecutable(), args, {
+      stdio: "inherit",
+    });
+  } catch (error) {
+    console.error(error);
+    process.exit(1);
```

Comment:
In JS, process.exit(…) should be avoided, in favour of process.exitCode = … . This last form let’s JS finish processing whatever it’s doing. It avoid aborting background process.